doozer/doozerlib/cli/scan_sources.py

```diff
@@ -54,6 +68,142 @@ def run(self):
         # We have our information. Now build and print the output report
         self.generate_report()
 
+    def _try_reconciliation(self, metadata: Metadata, repo_name: str, branch_name: str):
+        reconciled = False
+
+        # Attempt a fast-forward merge
+        rc, _, _ = exectools.cmd_gather(cmd=['git', 'pull', '--ff-only', 'public_upstream', branch_name])
+        if not rc:
+            # fast-forward succeeded, will push to openshift-priv
+            self.runtime.logger.info('Fast-forwarded %s from public_upstream/%s', metadata.name, branch_name)
+            reconciled = True
+
+        else:
+            # fast-forward failed, trying a merge commit
+            rc, _, _ = exectools.cmd_gather(cmd=['git', 'merge', f'public_upstream/{branch_name}',
```

Comment:
Would this be logging stdout/stderr? Would like that to help us investigate issue

Reply:
Updated